### ISSUE TITLE
Quick sort is implemented using mutable array

### ISSUE THREAD

**Issue by ashwinbhaskar** [ashwinbhaskar/functional-way#4]
The quick sort implemented in the folder Scala-Way is using a mutable array. Can/Should this be done in an immutable way?

**Comment by alex-morgun** [comment]
No, the interface should me immutable, local variables can be mutable for speed and efficiency. Not all problems are better solved pure FP

**Comment by alex-morgun** [comment]
But you can solve and make benchmark )

**Comment by ashwinbhaskar** [comment]
@genius-ua Agreed that not all problems are better solved with pure FP. But there are some data structures like braun trees and persistent vectors that can be used for immutability without compromising on performance
